app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -0,0 +1,148 @@
+                        else -> {
+                            Timber.d("Invalid state [player: $player, end_by: $endBy]")
+                            notifyEvent(Constants.EVENT_CANCELED, "'${mainActivity?.getString(R.string.external_player_unknown_error)}'")
+                        }
+                    }
+                }
+                Activity.RESULT_CANCELED -> {
+                    Timber.d("Playback stopped by user [player: $player]")
+                    notifyEvent(Constants.EVENT_CANCELED)
+                }
+                Activity.RESULT_FIRST_USER -> {
+                    Timber.d("Playback stopped by unknown error [player: $player]")
+                    notifyEvent(Constants.EVENT_CANCELED, "'${mainActivity?.getString(R.string.external_player_unknown_error)}'")
+                }
+                else -> {
+                    Timber.d("Invalid state [player: $player, resultCode: $resultCode]")
+                    notifyEvent(Constants.EVENT_CANCELED, "'${mainActivity?.getString(R.string.external_player_unknown_error)}'")
+                }
+            }
+            this.mainActivity = null
+        }
+
+        // https://wiki.videolan.org/Android_Player_Intents/
+        private fun handleVLCPlayer(resultCode: Int, data: Intent) {
+            val player = "VLC Player"
+            when(resultCode) {
+                Activity.RESULT_OK -> {
+                    val extraPosition = data.getLongExtra("extra_position", 0L)
+                    val extraDuration = data.getLongExtra("extra_duration", 0L)
+                    if(extraPosition > 0L) {
+                        Timber.d("Playback stopped [player: $player, extra_position: $extraPosition, extra_duration: $extraDuration]")
+                        notifyEvent(Constants.EVENT_TIME_UPDATE, extraPosition.toString())
+                        notifyEvent(Constants.EVENT_ENDED)
+                    } else {
+                        if(extraDuration == 0L && extraPosition == 0L) {
+                            Timber.d("Playback completed [player: $player]")
+                            notifyEvent(Constants.EVENT_TIME_UPDATE)
+                            notifyEvent(Constants.EVENT_ENDED)
+                        } else {
+                            Timber.d("Invalid state [player: $player, extra_position: $extraPosition, extra_duration: $extraDuration]")
+                            notifyEvent(Constants.EVENT_CANCELED, "'${mainActivity?.getString(R.string.external_player_unknown_error)}'")
+                        }
+                    }
+                }
+                else -> {
+                    Timber.d("Playback failed [player: $player, resultCode: $resultCode]")
+                    notifyEvent(Constants.EVENT_CANCELED, "'${mainActivity?.getString(R.string.external_player_unknown_error)}'")
+                }
+            }
+            this.mainActivity = null
+        }
+
+        private fun notifyEvent(event: String, parameters: String = "") {
```

Comment:
If you want, you could make this (and the other functions) extension functions of `MainActivity` to always have a context ready.

Reply:
I think it is not necessary to expose a function that will only be used in ExternalPlayer, the same with some constants, considering that the video player webapp events are eliminated in #104